contrib/glpk-cut-log.patch

```diff
@@ -86,39 +86,43 @@ index bbc8929..0bc64d1 100644
  AC_CONFIG_HEADERS([config.h])
  
 diff --git a/src/Makefile.am b/src/Makefile.am
-index b39efa6..2a025bc 100644
+index 48dff15..d04b547 100644
 --- a/src/Makefile.am
 +++ b/src/Makefile.am
-@@ -20,7 +20,10 @@ libglpk_la_LDFLAGS = \
- -version-info 36:0:1 \
- -export-symbols-regex '^glp_*'
+@@ -27,6 +27,7 @@ libglpk_la_LDFLAGS = \
+ ${NOUNDEFINED}
  
-+
-+# all of the cut log sources are listed first
  libglpk_la_SOURCES = \
 +cutlog01.c \
- glpapi01.c \
- glpapi02.c \
- glpapi03.c \
+ amd/amd_1.c \
+ amd/amd_2.c \
+ amd/amd_aat.c \
 diff --git a/src/cutlog01.c b/src/cutlog01.c
 new file mode 100644
-index 0000000..9a85bb7
+index 0000000..c8818df
 --- /dev/null
 +++ b/src/cutlog01.c
-@@ -0,0 +1,452 @@
+@@ -0,0 +1,459 @@
 +/* cutlog01.c (api extension routines) */
 +
++#undef NEW_LOCAL
 +
-+#include "glpapi.h"
-+#include "glpios.h"
++#include "env.h"
++#include "ios.h"
++#include "npp.h"
++#include "glpk.h"
++#undef NEW_LOCAL
++#include "ios.h"
 +
++#if 0
 +int glp_get_it_cnt(glp_prob *P){
 +  if(P == NULL){
 +    return 0;
 +  }else{
 +    return P->it_cnt;
 +  }
 +}
++#endif
```

Comment:
Maybe add a comment here why this is not removed but kept.

Reply:
This was a totally valid point, and should have been removed.

GLPK 4.53 added `glp_get_it_cnt` as a native API, which means Tim's version could be removed:

>       Two API routines glp_get_it_cnt and glp_set_it_cnt were added.
>       Thanks to Joey Rios <joeylrios@hotmail.com> for suggestion.
>

[https://fossies.org/linux/glpk/ChangeLog]